assignment_A1.ipynb

```diff
@@ -426,33 +426,24 @@
     "        \n",
     "    def smallest(self):\n",
     "        \"\"\"Return node with smallest value\"\"\"\n",
-    "        sml  = Node(float(\"inf\"))\n",
-    "        currentNode = self.__header\n",
-    "        for i in range(self.get_size()+2):  #traverse from header upto (and including) trailer \n",
-    "            cur = currentNode.get_data()\n",
-    "            sml_data=sml.get_data()\n",
-    "            if type(cur)!=str:\n",
-    "                if cur < sml_data:\n",
-    "                    currentNode=sml\n",
-    "                    currentNode = currentNode.get_next()\n",
-    "                else:\n",
-    "                    currentNode = currentNode.get_next()\n",
-    "            else:\n",
-    "                currentNode = currentNode.get_next()\n",
-    "                if currentNode==None:\n",
-    "                    break\n",
+    "        sml  = float(\"inf\")\n",
+    "        currentNode = self.get_first()\n",
+    "        for i in range(self.get_size()): #traverse from first node to last node \n",
+    "            x = currentNode.get_data()\n",
+    "            if isinstance(x, (int, float)):\n",
+    "                if x < sml:\n",
+    "                    sml = x\n",
+    "            currentNode = currentNode.get_next()\n",
+    "            \n",
     "        return sml\n",
     "\n",
     "    \n",
     "    def traverse(self, visitor):\n",
     "        \"\"\"Visit every element in the list, call visitor\"\"\"\n",
     "        currentNode = self.__header\n",
-    "        for i in range(self.get_size()+2): #traverse from header upto (and including) trailer \n",
```

Comment:
Bij nader inzien is het helemaal niet logisch om de traverse langs header en trailer te laten gaan